File: source/code/csharp/authentication/create_asm.cs
var application = await client.GetApplicationAsync(
    "https://api.stormpath.com/v1/applications/1gk4Dxzi6o4PbdleXaMPL");

var directory = await client.GetDirectoryAsync(
    "https://api.stormpath.com/v1/directories/2SKhstu8PlaekcaEXampLE");

var newMapping = client.Instantiate<IApplicationAccountStoreMapping>()
    .SetApplication(application)
    .SetAccountStore(directory)
    .SetListIndex(0)
    .SetDefaultAccountStore(true)
    .SetDefaultGroupStore(true);

await application.CreateAccountStoreMappingAsync(newMapping);

File: source/code/java/authentication/create_asm.java
ApplicationAccountStoreMapping mapping = client.instantiate(ApplicationAccountStoreMapping.class)
    .setAccountStore(accountStore) // this could be an existing group or a directory
    .setApplication(application)
    .setDefaultAccountStore(Boolean.TRUE)
    .setDefaultGroupStore(Boolean.TRUE)
    .setListIndex(0);

application.createAccountStoreMapping(mapping);
